subcommand.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+)
+
+type SubCommand struct {
+	cmd      *exec.Cmd
+	PipeRead *os.File
+}
+
+type Stream int
+
+const (
+	StreamStdout Stream = 1 << iota
+	StreamStderr
+)
+
+func NewSubCommand(args []string, stream Stream) SubCommand {
+	pipeRead, pipeWrite, err := os.Pipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	env := append(os.Environ(), "GODEBUG=gctrace=1")
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = env
+	cmd.Stdin = os.Stdin
+
+	switch stream {
+	case StreamStdout:
+		cmd.Stdout = pipeWrite
```

Comment:
I'm not sure I'm understanding this comment. The only difference between master and this branch is the switch statement (and I only added it to make it testable). In `main.go`, I'm still using `StreamStderr`, which means this should behave the same as master.

Speaking of failing commands, I can see now a race condition where `s.cmd.Run()` fails and `log.Fatal(err)` exits before the parser has time to finish its job. But that would apply to the master branch too, wouldn't it?

I will have another run to add tests around this scenario.